SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -27,58 +27,177 @@ This SWIP is part of a series of SWIPs (but can be implemented on its own). To s
+* Resolving the conversion from honey to money.
 * Ensuring that the user can engage in SWAP accounting for the chosen payment module before payment is due.
-* Sending the recipient a payment.
-* Returning true when the payment was successful.
-* Returning false when the payment was not successful.
-* Verifying any on-chain conditions and making sure that they don't change within a reasonable time window after the transaction happens (due to block reorganizations / payment-module specific conditions)
-* Referencing a type, version and base currency 
+* Sending a payment to the recipient.
+* Sending a message to the accounting module upon receipt of payment
+* Ensuring that the user can send payments with the chosen payment module before payment is due.
+* Handling any potential errors.
 * Optionally exposing other methods such as querying balances, topping up balances or sending payments (outside of Swarm). 
 
-Nodes can specify their preference for both payment module, as well as price oracle in a list in a configuration file. The preferences are normalized and weighed.
-For any preference list, the chosen option will be the option which has the highest cumulative preference. The preference list has three dimensions, which will be resolved from high to low:
+### Allowing multiple payment modules
+When allowing multiple payment modules, it is essential for nodes to communicate their preference. There are three dimensions to take into consideration for the payment preference: 
 
 1. Currency to use
 2. Price oracle to use
-3. Payment method to use
+3. Payment provider to use
+
+We propose a standardized way to communicate these preferences and an algorithm to resolve any two preferences lists.
+
+For any dimension, nodes list the relative importance of them. Consequently, nodes specify their accepted options within each dimension. Importance is communicated through Swarm by any positive integer.
+
+The following is an example of the proposed configuration:
+
+```yaml
+# The relative importance of each dimension for the node
+dimensions:
+	currency: 70
+	provider: 20
+	oracle: 10
+
+# Supported payment options by the node.
+# A payment module is defined by a currency, provider and oracle. Weights are the associated preferences.
+currencies:
+	rif:
+		weight: 36
+		providers:
+			lumino:
+				weigth: 15
+			raiden:
+				weight: 45
+		oracles:
+			rifOracleA:
+				weigth: 10
+			rifOracleB:
+				weigth: 70
+			rifOracleC:
+				weigth: 60
+	xdai:
+		weight: 9
+		providers:
+			Lumino: 
+				weigth: 10
+		oracles:
+			xDaiOracle: 
+				weight: 20
+	dai:
+		weight: 5
+		providers:
+			raiden:
+				weight: 80
+		oracles:
+			daiOracleA: 
+				weight: 30
+			daiOracleB:
+				weight: 20
+	eth:
+		weight: 40
+		providers:
+			swap:
+				weight: 35
+		oracles:
+			ethOracleA:
+				weight: 90
+			ethOracleB:
+				weight: 10
+```
+
+This configuration is used by the payment method selection algorithm in the following way: 
+
```

Comment:
We should introduce node A and node B and say that they both independently execute the steps below. The only point where the nodes interact is in step 2.

Reply:
You are right. I added the sentence before describing the steps of the algorithm.